openedx/features/partners/forms.py

```diff
@@ -0,0 +1,14 @@
+from django.utils.translation import ugettext as _
+
+from student.forms import PasswordResetFormNoActive
+
+
+class PartnerResetPasswordForm(PasswordResetFormNoActive):
+    """
+    A form to validate reset password data for partner users. This is currently only
+    being used to override an error message as per the requirements.
+    """
+    def clean_email(self):
+        self.error_messages['unknown'] = _("We don't recognize the email: {}").format(self.cleaned_data["email"])
+        super(PartnerResetPasswordForm, self).clean_email()
```

Comment:
Are you sure it is working, ideally it should be `return super(PartnerResetPasswordForm, self).clean_email()`, since `clean_email()` in parent class is returning email